File: command/Ex1/LightCommand.java
public class LightCommand extends CommandBase {
    private Light light;

    public LightCommand(Light light) {
        this.light = light;
    }

    @Override
    public void executeOn() {
        this.light.on();
        this.state = CommandState.ON;
    }

    @Override
    public void executeOff() {
        this.light.off();
        this.state = CommandState.OFF;
    }

    @Override
    public void undo() {
        if(this.state == CommandState.ON) {
            this.executeOff();
        } else {
            this.executeOn();
        }
    }

    @Override
    public void redo() {
        if (this.state == CommandState.ON) {
            this.executeOn();
        } else {
            this.executeOff();
        }
    }
}

File: command/Ex1/Ex1/LightCommand.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Ex1
{
    public class LightCommand : CommandBase
    {
        private Light light;

        public LightCommand(Light light)
        {
            this.light = light;
        }

        public override void ExecuteOn()
        {
            this.light.On();
            this._state = CommandState.On;
        }

        public override void ExecuteOff()
        {
            this.light.Off();
            this._state = CommandState.Off;
        }

        public override void ExecuteUndo()
        {
            if (this._state == CommandState.On)
            {
                this.ExecuteOff();
                this._state = CommandState.Off;
            }
            else
            {
                this.ExecuteOn();
                this._state = CommandState.On;
            }
        }

        public override void ExecuteRedo()
        {
            if (this._state == CommandState.On)
            {
                this.ExecuteOn();
                this._state = CommandState.On;
            }
            else
            {
                this.ExecuteOff();
                this._state = CommandState.Off;
            }
        }
    }
}
